Fix Go discovery bugs

saicheems opened the pull request.

broady commented on src/test/java/com/google/api/codegen/testdata/discoveries/go/go_appengine.v1beta5.json.baseline:
```diff
@@ -737,7 +737,7 @@ func main() {
   }
 
-  resp, err := s.Apps.Services.Versions.Patch(appsId, servicesId, versionsId, &version).Context(ctx).Do()
+  resp, err := s.Apps.Services.Versions.Patch(appsId, servicesId, versionsId, version).Context(ctx).Do()
```
Can we map these names to `appsID`? no big deal if not.

saicheems commented:
Sorry I think I reuploaded while you were reviewing

Removed `fmt` in favor of `log` in every case, and renamed the request body to `body` to avoid some frequent name clashes.

saicheems commented:
As for the `Id` thing... It's worth making note of

It's a bit tricky to detect acronyms, but we might be able to special case common ones.

broady commented:
yeah, don't bother with the `Id` thing.

saicheems commented:
Renamed `body` to `resource`

broady commented:
LGTM
